core/services/synchronization/telemetry_ingress_batch_client.go

```diff
@@ -106,12 +113,28 @@ func (tc *telemetryIngressBatchClient) Start(ctx context.Context) error {
 		// Initialize a new wsrpc client caller
 		// This is used to call RPC methods on the server
 		if tc.telemClient == nil { // only preset for tests
-			conn, err := wsrpc.DialUniWithContext(ctx, tc.lggr, tc.url.String(), clientPrivKey, serverPubKey)
-			if err != nil {
-				return fmt.Errorf("Could not start TelemIngressBatchClient, Dial returned error: %v", err)
+			if tc.useUniConn {
+				go func() {
+					// Use background context to retry forever to connect
+					// Blocks until we connect
+					conn, err := wsrpc.DialUniWithContext(ctx, tc.lggr, tc.url.String(), clientPrivKey, serverPubKey)
+					if err != nil {
+						tc.lggr.Criticalw("telemetry endpoint dial errored unexpectedly", "err", err)
```

Comment:
If the node doesn't ever connect to the backend (bad creds, for example) and we gracefully shutdown, this will fire. Any harm in that? I'm not familiar with `Critical` on the logger

Reply:
Hm no harm, but perhaps now that we're using the app ctx just making it a warn makes more sense

Reply:
No harm per se. `critical` is just a more sever level than `error` that we expect node ops to respond to. That being said, we could maybe skip for `ctx.Err()`? Is that how graceful shutdown presents here?

Reply:
lol